src/MatrixFields/band_matrix_row.jl

```diff
@@ -0,0 +1,153 @@
+"""
+    BandMatrixRow{ld}(entries...)
+
+Stores the nonzero entries in a row of a band matrix, starting with the lowest
+diagonal, which has index `ld`. Supported operations include accessing the entry
+on the diagonal with index `d` by calling `row[d]`, taking linear combinations
+with other band matrix rows, and checking for equality with other band matrix
+rows. There are several aliases defined for commonly-used subtypes of
+`BandMatrixRow` (with `T` denoting the type of the row's entries):
+- `DiagonalMatrixRow{T}`
+- `BidiagonalMatrixRow{T}`
+- `TridiagonalMatrixRow{T}`
+- `QuaddiagonalMatrixRow{T}`
+- `PentadiagonalMatrixRow{T}`
+"""
+struct BandMatrixRow{ld, bw, T} # bw is the bandwidth (the number of diagonals)
+    entries::NTuple{bw, T}
+end
+BandMatrixRow{ld}(entries::Vararg{Any, bw}) where {ld, bw} =
+    BandMatrixRow{ld, bw}(entries...)
+function BandMatrixRow{ld, bw}(entries::Vararg{Any, bw}) where {ld, bw}
+    promoted_entries = promote(entries...)
+    return BandMatrixRow{ld, bw, eltype(promoted_entries)}(promoted_entries)
+end
+
+const DiagonalMatrixRow{T} = BandMatrixRow{0, 1, T}
+const BidiagonalMatrixRow{T} = BandMatrixRow{-1 + half, 2, T}
+const TridiagonalMatrixRow{T} = BandMatrixRow{-1, 3, T}
+const QuaddiagonalMatrixRow{T} = BandMatrixRow{-2 + half, 4, T}
+const PentadiagonalMatrixRow{T} = BandMatrixRow{-2, 5, T}
+
+function Base.show(
+    io::IO,
+    ::Type{BMR},
+) where {ld, bw, T, BMR <: BandMatrixRow{ld, bw, T}}
+    string = if BMR <: DiagonalMatrixRow
+        "DiagonalMatrixRow{$T}"
+    elseif BMR <: BidiagonalMatrixRow
+        "BidiagonalMatrixRow{$T}"
+    elseif BMR <: TridiagonalMatrixRow
+        "TridiagonalMatrixRow{$T}"
+    elseif BMR <: QuaddiagonalMatrixRow
+        "QuaddiagonalMatrixRow{$T}"
+    elseif BMR <: PentadiagonalMatrixRow
+        "PentadiagonalMatrixRow{$T}"
+    else
+        "BandMatrixRow{$ld, $bw, $T}"
+    end
+    print(io, string)
+end
```

Comment:
Unfortunately, overloading `Base.show` for `::Type{T}` invalidates `show` in Base. Can we change this to `Base.show(io::IO, ::BandMatrixRow)` instead?

Reply:
The goal here was actually to change the way matrix fields are printed, rather than individual rows. For example:
```
julia> ᶠᶠmat
TridiagonalMatrixRow{Float64}-valued Field:
  entries: 
    1: [0.667658, 0.766307, 0.866266, 0.0461284, 0.439407, 0.390036, 0.0762841, 0.878829, 0.0061787, 0.0188659  …  0.398943, 0.871208, 0.0441692, 0.878441, 0.982707, 0.0220367, 0.712676, 0.0229924, 0.560176, 0.692064]
    2: [0.95014, 0.690321, 0.463986, 0.189937, 0.00750967, 0.00224279, 0.5567, 0.830047, 0.527776, 0.519096  …  0.65025, 0.284222, 0.0891259, 0.00904144, 0.115214, 0.612847, 0.625339, 0.38462, 0.0864744, 0.369243]
    3: [0.503339, 0.620339, 0.343298, 0.918134, 0.0660757, 0.776861, 0.3916, 0.0791701, 0.496441, 0.0822908  …  0.735152, 0.521021, 0.970042, 0.0290453, 0.650956, 0.53432, 0.610975, 0.368205, 0.930298, 0.133497]
```
Should I just directly overwrite `Base.show(io::IO, ::Field)` instead?

Reply:
> Should I just overwrite ...

Yeah, that's probably a lot better